### ISSUE TITLE
Allow access to set JSESSIONID to HTTPOnly or to prevent it being set at all

### ISSUE THREAD

**Issue by elpete** [cfmlprojects/runwar#100]
Currently, the `JSESSIONID` is automatically sent to the browser, but it is sent as `HTTPOnly = false`.  Please provide an argument to enable `HTTPOnly = true` for the `JSESSONID` as well as disabling the cookie being set entirely.

Linked CommandBox issue: https://ortussolutions.atlassian.net/browse/COMMANDBOX-724

**Comment by denuno** [comment]
Hmm, I'll have to look into what options there are.  Theoretically we could add a web.xml
```
<session-config>
      <cookie-config>
        <http-only>true</http-only>
        <secure>true</secure>
      </cookie-config>
    </session-config>
```
type of thing on the fly, so that'd do the secure cookies (might also want to toss in no url-based sessions :smiley:  ) but suppressing jsessionid might be tricky.

**Comment by elpete** [comment]
Setting it `HTTPOnly` and `secure` addresses any security concerns. Suppressing it would be bonus.

**Comment by lmajano** [comment]
If we can avoid a custom `web.xml` that is my preference. If we can do this via settings like we do with all settings, that is a preference.

**Comment by bdw429s** [comment]
@lmajano I think we can do both.  The `web.xml` stuff @denuno referenced is actually a standard spec and would be useful to support for people wanting to run a war directly on runwar/CommandBox that used that setting.  What we can do is inject our own setting to override what's in the web.xml which is what we're already doing for things like welcome files.  Since we parse the web.xml we have full control over it.

**Comment by denuno** [comment]
Indeed!  Handled the same as welcomefiles.  I'm thinking `secure-cookies` that sets both the httpOnly and secure flags initially, eventually maybe splitting them out.  Though if we do split them out, I like prefixing the options similarly so they group/order, so it'd be `securecookies-httponly... I'd like `cookies-httponly` better than that but `cookies-secure` is a bit meh.  We might do other cookie stuff tho so leaning a bit harder towards a `cookies` prefix.

**Comment by bdw429s** [comment]
@denuno Can you confirm this is just a single setting that controls both the secure and HTTP only aspect of the default session cookie?

**Comment by denuno** [comment]
Yeah, just the one, if you've got a reason for why it should be two we can split 'em tho.

**Comment by bdw429s** [comment]
Well, I think of the obvious reason would be if a user wants to have a secure session cookie that's accessible via JS on the page, or if a user wanted a cookie to be HTTP only but wanted to pass it over HTTP.  We're assuming that those scenarios will never be asked for.  Perhaps that's a safe assumption, but I don't know.

**Comment by denuno** [comment]
Meh, as I mentioned, I like `cookie` better for a prefix if we split them out.  It was more of a "here's the common one used to pass verification checks" than a prediction of what would be wanted.  Really these settings should be handled/verified by the application itself for safety.

Since there will be environments/profiles for handling multiple settings in one setting, ish, probably better to go ahead and nix the `securecookies` and split it out into two settings.  Then just set them both to true for the `production` one or something.

**Comment by bdw429s** [comment]
I like that.  Let me know when you have it broken out and I'll test it.

**Comment by bdw429s** [comment]
@denuno Did the latest snapshot break out these settings?

**Comment by bdw429s** [comment]
@denuno Will these settings be broken out as part of 3.8.1?

**Comment by denuno** [comment]
Yep, I splitted 'em!  Kept the easy single on tho. :smile:

**Comment by bdw429s** [comment]
@denuno Can you document the final names of the new config settings here please?

**Comment by bdw429s** [comment]
@denuno Can you document the final names of the new config settings here please?

**Comment by denuno** [comment]
```
  -cookiehttponly,--cookie-httponly <true|false>
    Set cookie 'http-only' header

  -cookiesecure,--cookie-secure <true|false>
    Set cookie 'secure' header
```

**Comment by bdw429s** [comment]
Thx @denuno , I'll plug those in!

**Comment by bdw429s** [comment]
@denuno This doesn't seem to be working.  I set the following:
```
--cookie-secure true
--cookie-httponly true
```
and enabled session type of `JEE` in my Lucee server so I would get a `JSessionID` but the cookie is not set to HTTP only.  Note I didn't enable SSL for this first test.

I reviewed the code in Runwar and it only seems half way implemented.  I'm not even sure how it should be working.  I can only find this single reference in `Server.java`

```
        if(serverOptions.isSecureCookies()) {
                sessionConfig.setHttpOnly(true);
                sessionConfig.setSecure(true);
        }
```
I'm not even using that option name.  So far as I can tell, there is no code checking the server options to see if `isCookieHttpOnly()` and/or `isCookieSecure()` are set.

**Comment by denuno** [comment]
@bdw429s Thanks!  That was it.  Should be fixed!

**Comment by bdw429s** [comment]
@denuno Can you me the steps you've been using to test.  I'm not sure if I'm doing it right but I just can't seem to get this to work.  Here's what I did:
```
server set app.sessionCookieHTTPOnly=true
server start
```
This  spins up a Lucee 5 server with the JVM args
```
 --cookie-secure false
 --cookie-httponly true
```
When I open up the web browser and my Chrome tools, the `JSESSIONID` cookie does not have a checkmark under the "HTTP" column.  I've tried switching the Lucee setting for session variables from "application" to "JEE" but it doesn't seem to make a difference.

The same is true for Adobe server but I can't access the admin right now to change the settings due to the most recent runwar regression on that.

**Comment by bdw429s** [comment]
@denuno Did this ever get fixed?  I never heard back after my last report above.
